buoy.php

```diff
@@ -107,13 +107,24 @@ public static function registerL10n () {
      * @return void
      */
     public static function initialize () {
-        // Make sure the WP REST API plugin is installed.
-        // We can remove this if block once that plugin is merged
-        // into WP Core.
-        require_once ABSPATH.'wp-admin/includes/plugin.php';
-        if (is_plugin_inactive('rest-api/plugin.php')) {
-            if (is_wp_error(activate_plugin('rest-api/plugin.php'))) {
-                self::install_plugin_dependency('rest-api');
+        // WordPress versions prior to 4.7 did not fully merge the
+        // WP REST API, on which we (partially) rely. In version 4.7,
+        // however, the REST API's important parts for our use case,
+        // the so-called "Content API" did indeed land in core. This
+        // means we no longer need to force-install it ourselves.
+        //
+        // See https://codex.wordpress.org/Version_4.7 highlights.
+        //
+        // For now, make sure the WP REST API plugin is installed and
+        // activated for users still running older WordPress versions.
+        // We can remove this check once enough Buoys update to v4.7.
+        global $wp_version;
+        if (version_compare('4.7', $wp_version) === 1) {
```

Comment:
My understanding of `version_compare()` has always been flaky because I generally just don't do version comparison. That said, it seems to me like the PHP manual suggests that in the case you describe where a user updates to WP > 4.7, the evaluated code would be: `version_compare('4.7', '4.8') === 1`.

Since 4.8 is greater than 4.7, I think the function would a `-1` (i.e., according to the PHP manual, "By default, version_compare() returns `-1` if the first version is lower than the second") and thus the condition will be `false`. In this case that is what we want because WP version 4.8 also is expected to include the WP REST API in core and so we shouldn't install it ourselves.

Similarly, if the user installs Buoy *for the first time* on a site running WordPress 4.7, this function would return a `0`, and similarly not cause a needless install of the WP REST API plugin.

In the case where a user installs Buoy on a site running WordPress < 4.7, though, such as 4.6, then the first version string would be bigger than the second, and thus the function would return exactly a `1`, because as the manual says the function returns:

>  `1` if the second [argument version string] is lower.

That's the behavior we want because it is only those pre-4.7 WordPress installs that we want to include this neccessary plugin for.

Is my logic sound?